pkg/metrics/queue.go

```diff
@@ -57,6 +58,8 @@ type QueueMetrics struct {
 	resourceMetricsLabel *prometheus.GaugeVec
 	// Deprecated - To be removed in 1.7.0. Replaced with queue label Metrics
 	resourceMetricsSubsystem *prometheus.GaugeVec
+	// Track known resource types.
+	knownResourceTypes map[string]struct{}
```

Comment:
Need a lock here to guard access to knownResourceTypes.

Reply:
Addressed in latest PR, thanks!